src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/DoNotGuardDictionaryRemoveByContainsKeyTests.cs

```diff
@@ -0,0 +1,298 @@
+	            MyDictionary.Remove(""Key"");
+        }
+    }
+}";
+
+            await VerifyCS.VerifyAnalyzerAsync(source);
+        }
+
+        [Fact]
+        public async Task RemoveInSeparateLine_NoDiagnostic_CS()
+        {
+            string source = @"
+" + CSUsings + @"
+namespace Testopolis
+{
+    public class MyClass
+    {
+        private readonly Dictionary<string, string> MyDictionary = new Dictionary<string, string>();
+
+        public MyClass()
+        {
+            if (MyDictionary.ContainsKey(""Key""))
+	            _ = MyDictionary.Count;
+	        MyDictionary.Remove(""Key"");
+        }
+    }
+}";
+
+            await VerifyCS.VerifyAnalyzerAsync(source);
+        }
+
+        [Fact]
+        public async Task AdditionalStatements_NoDiagnostic_CS()
+        {
+            string source = @"
+" + CSUsings + @"
+namespace Testopolis
+{
+    public class MyClass
+    {
+        private readonly Dictionary<string, string> MyDictionary = new Dictionary<string, string>();
+
+        public MyClass()
+        {
+            if (MyDictionary.ContainsKey(""Key""))
+            {
+                MyDictionary.Remove(""Key"");
+                Console.WriteLine();
+            }
+        }
+    }
+}";
+
+            await VerifyCS.VerifyAnalyzerAsync(source);
+        }
+
+        [Fact]
+        public async Task RemoveIsTheOnlyStatement_ReportsDiagnostic_VB()
+        {
+            string source = @"
+" + VBUsings + @"
+Namespace Testopolis
+    Public Class SomeClass
+        Public MyDictionary As New Dictionary(Of String, String)()
+
+        Public Sub New()
+            If [|MyDictionary.ContainsKey(""Key"")|] Then
+                MyDictionary.Remove(""Key"")
+            End If
+        End Sub
+    End Class
+End Namespace";
+
+            string fixedSource = @"
+" + VBUsings + @"
+Namespace Testopolis
+    Public Class SomeClass
+        Public MyDictionary As New Dictionary(Of String, String)()
+
+        Public Sub New()
+            MyDictionary.Remove(""Key"")
+        End Sub
+    End Class
+End Namespace";
+
+            await VerifyVB.VerifyCodeFixAsync(source, fixedSource);
+        }
+
+        [Fact]
+        public async Task AdditionalStatements_NoDiagnostic_VB()
+        {
+            string source = @"
+" + VBUsings + @"
+Namespace Testopolis
+    Public Class SomeClass
+        Public MyDictionary As New Dictionary(Of String, String)()
+
+        Public Sub New()
+            If MyDictionary.ContainsKey(""Key"") Then
+                MyDictionary.Remove(""Key"")
+                Console.WriteLine()
+            End If
+        End Sub
+    End Class
+End Namespace";
+
+            await VerifyVB.VerifyAnalyzerAsync(source);
+        }
+        #endregion
+
+        #region Helpers
+        private const string CSUsings = @"using System;
+using System.Collections.Generic;";
+
+        private const string VBUsings = @"Imports System
+Imports System.Collections.Generic";
+        #endregion
+    }
+}
```

Comment:
The test seems did not cover a case with `out` variable
```cs
    if (data.ContainsKey(key))
        data.Remove(key, out var value)
```
Probably this will not be used with a single remove statement, but i think still good to have

Reply:
I've added a test. This overload is new in .NET Core; am I using the correct syntax to skip running the test for .NET Framework?

```csharp
            await new VerifyCS.Test
            {
                TestCode = source,
                FixedCode = fixedSource,
                ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp31
            }.RunAsync();
```

Reply:
That should be correct, by my understanding `ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp31` will make the test run only on `netcoreapp3.1` but that is enough as i don't think there is an option to say not run on  `.NET Framework`